code/modules/clothing/modular/armor_pieces.dm

```diff
@@ -0,0 +1,148 @@
+/**
+	Modular armor pieces
+
+	These are straight armor attachments that equip into 1 of 3 slots on modular armor
+	There are chest arms and leg variants, each with their own light medium and heavy armor (not mentioning special types)
+
+	Each armor will merge its armor with the modular armor on attachment and remove it when detached, similar with other stats like slowdown.
+*/
+/obj/item/armor_module/armor
+	name = "modular armor - armor module"
+	icon = 'icons/mob/modular/modular_armor.dmi'
+
+	/// The additional armor provided by equipping this piece.
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 0, "rad" = 0, "fire" = 0, "acid" = 0)
+
+	/// Addititve Slowdown of this armor piece
+	slowdown = 0
+
+/obj/item/armor_module/armor/do_attach(mob/living/user, obj/item/clothing/suit/modular/parent)
+	. = ..()
+	parent.armor = parent.armor.attachArmor(armor)
+	parent.slowdown += slowdown
+
+/obj/item/armor_module/armor/do_detach(mob/living/user, obj/item/clothing/suit/modular/parent)
+	parent.armor = parent.armor.detachArmor(armor)
+	parent.slowdown -= slowdown
+	return ..()
+
+
+
+/** Chest pieces */
+/obj/item/armor_module/armor/chest
+	icon_state = "medium_chest_icon"
+	item_state = "medium_chest"
+
+/obj/item/armor_module/armor/chest/do_attach(mob/living/user, obj/item/clothing/suit/modular/parent)
+	. = ..()
+	parent.slot_chest = src
+	parent.update_overlays()
+
+/obj/item/armor_module/armor/chest/do_detach(mob/living/user, obj/item/clothing/suit/modular/parent)
+	parent.slot_chest = null
+	return ..()
```

Comment:
Shouldn't both update overlays?

Reply:
True, because its removal the overlay update is last and called by the parent since the parent is called last.

On attach we call the parent first so the overlay might be too soon, which is why i manually delay it